### PR TITLE
RD-5247 automatically advance next in Getting Started Modal

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Jira ticket: RD-5247

## Screenshots / Videos

https://user-images.githubusercontent.com/17104212/185927133-c9e162f3-3103-4e2e-8d04-758827cd7108.mov

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
`getting_started_spec`:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3451/pipeline

 

## Documentation
There is no Getting Started Modal in the documentation at all. So, there is nothing to update.
https://docs.cloudify.co/latest/working_with/console/getting-started/

### CODE REVIEW

**Comment by leon-good-life** [comment]
I started `getting_started_spec`:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3451/pipeline

**Comment by leon-good-life** [comment]
The test failed. I will fix it.

**Comment by leon-good-life** [comment]
I started `getting_started_spec:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3472/pipeline

File: app/components/GettingStartedModal/GettingStartedModal.tsx [hunk 1]
```diff
@@ -95,6 +99,8 @@ const GettingStartedModal = () => {
 
     const handleEnvironmentsStepChange = (selectedEnvironments: GettingStartedEnvironmentsData) => {
         setEnvironmentsStepData(selectedEnvironments);
+        goToNextStep();
+        setSecretsStepIndex(0);
```
Comment (leon-good-life) [inline app/components/GettingStartedModal/GettingStartedModal.tsx:103]: ** Development note: **
In L101, `setEnvironmentsStepData(selectedEnvironments);` is updating the state.
In my previous implementation I was calling `handleNextClick()` in L102.
L102 is called before the state updated. I verified it by using `console.log`.
So I changed the implementation.

File: app/components/GettingStartedModal/GettingStartedModal.tsx [hunk 2]
```diff
@@ -95,6 +99,8 @@ const GettingStartedModal = () => {
 
     const handleEnvironmentsStepChange = (selectedEnvironments: GettingStartedEnvironmentsData) => {
```
Comment (kubama) [inline app/components/GettingStartedModal/GettingStartedModal.tsx:100]: I think `handleEnvironmentClick` would be a more appropriate name for this function after this change.

**Comment by leon-good-life** [comment]
I fixed system tests and run Jenkins again:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3476/pipeline

File: app/components/GettingStartedModal/GettingStartedModal.tsx (hunk 2 above)
Comment (leon-good-life) [reply app/components/GettingStartedModal/GettingStartedModal.tsx]: I addressed it: [2b27990](https://github.com/cloudify-cosmo/cloudify-stage/pull/2134/commits/2b27990b32ff3871ced19f370026373fecf68254)

**Comment by leon-good-life** [comment]
I updated system test and started jenkins again:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3477/pipeline

File: test/cypress/integration/getting_started_spec.ts [hunk 3]
```diff
@@ -330,16 +326,6 @@ describe('Getting started modal', () => {
             });
         });
 
-        it('should allow only one environment to be selected at a time', () => {
```
Comment (leon-good-life) [inline test/cypress/integration/getting_started_spec.ts:333]: This test is irrelevant now

File: test/cypress/integration/getting_started_spec.ts [hunk 4]
```diff
@@ -330,31 +326,11 @@ describe('Getting started modal', () => {
...
-            goToNextStep();
-            cy.contains('button', 'AWS').click();
-            cy.contains('button.active', 'AWS');
-
-            cy.contains('button', 'GCP').click();
-            cy.contains('button.active', 'AWS').should('not.exist');
-            cy.contains('button.active', 'GCP');
-        });
-
-        it('should allow to click "Next" button when an environment is being selected', () => {
```
Comment (leon-good-life) [inline test/cypress/integration/getting_started_spec.ts:343]: This test is irrelevant now

**Comment by leon-good-life** [comment]
Te test passed

**Review by kubama — APPROVED** [review]

**Pull request merged**